--- origsrc/tapioca-sharp/configure.ac	2007-06-08 15:01:00.000000000 -0500
+++ src/tapioca-sharp/configure.ac	2007-06-08 15:12:41.465371000 -0500
@@ -17,12 +17,20 @@
   AC_MSG_ERROR([You need to install pkg-config])
 fi
 
-dnl Find pkg-config
-AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-if test "x$PKG_CONFIG" = "xno"; then
-        AC_MSG_ERROR([You need to install pkg-config])
+
+dnl -- The GAC Tool
+
+AC_PATH_PROG(GACUTIL, gacutil, no)
+if test "x$GACUTIL" = "xno" ; then
+	AC_MSG_ERROR([No gacutil tool found])
 fi
 
+AC_SUBST(GACUTIL)
+
+GACUTIL_FLAGS='/gacdir $(DESTDIR)$(libdir)'
+AC_SUBST(GACUTIL_FLAGS)
+
+
 dnl Find Mono and Gmcs
 MONO_REQUIRED_VERSION=1.1.17
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
--- origsrc/tapioca-sharp/tapioca/Makefile.am	2007-06-08 15:00:59.000000000 -0500
+++ src/tapioca-sharp/tapioca/Makefile.am	2007-06-08 15:11:27.261296200 -0500
@@ -2,8 +2,9 @@
 
 ASSEMBLY_NAME = INdT.Tapioca
 ASSEMBLY = $(ASSEMBLY_NAME).dll
-INdT_Tapiocadir = $(pkglibdir)
+INdT_Tapiocadir = $(libdir)/mono/$(PACKAGE)
 INdT_Tapioca_SCRIPTS = $(ASSEMBLY)
+SNK=$(top_srcdir)/tapioca-sharp.snk
 
 TARGET = $(ASSEMBLY)
 
@@ -45,7 +46,16 @@
 
 
 $(ASSEMBLY): $(SOURCES) $(UTILSOURCES)
-	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $(SOURCES) $(UTILSOURCES) $(PKG_REFERENCES)
+	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $^ $(PKG_REFERENCES) -keyfile:$(SNK)
 
 all: $(ASSEMBLY)
 
+### GAC install rules ###
+
+install-data-local:
+	echo "$(GACUTIL) /i $(ASSEMBLY) $(GACUTIL_FLAGS)"; \
+	$(GACUTIL) /i $(ASSEMBLY) $(GACUTIL_FLAGS) || exit 1;
+
+uninstall-local:
+	echo "$(GACUTIL) /u $(ASSEMBLY_NAME) $(GACUTIL_FLAGS)"; \
+	$(GACUTIL) /u $(ASSEMBLY_NAME) $(GACUTIL_FLAGS) || exit 1;
--- origsrc/tapioca-sharp/tapioca-sharp.pc.in	2007-06-08 15:01:00.000000000 -0500
+++ src/tapioca-sharp/tapioca-sharp.pc.in	2007-06-08 15:02:48.551531800 -0500
@@ -7,4 +7,4 @@
 Version: @VERSION@
 Cflags:
 Requires: telepathy-sharp
-Libs: -r:${libdir}/tapioca-sharp/INdT.Tapioca.dll
+Libs: -r:${libdir}/mono/tapioca-sharp/INdT.Tapioca.dll
--- origsrc/tapioca-sharp/tests/Makefile.am	2007-06-08 15:00:59.000000000 -0500
+++ src/tapioca-sharp/tests/Makefile.am	2007-06-08 15:03:50.989831000 -0500
@@ -22,6 +22,6 @@
 all: $(TARGET)
 
 $(TARGET): $(ASSEMBLY_SOURCES)
-	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $(ASSEMBLY_SOURCES) \
+	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $^ \
 	-target:exe -out:"$@"
 
