diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/tapioca-sharp/configure.ac src/tapioca-sharp/configure.ac
--- origsrc/tapioca-sharp/configure.ac	2007-06-08 15:01:00.000000000 -0500
+++ src/tapioca-sharp/configure.ac	2007-06-08 15:12:41.465371000 -0500
@@ -17,12 +17,20 @@
   AC_MSG_ERROR([You need to install pkg-config])
 fi
 
-dnl Find pkg-config
-AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-if test "x$PKG_CONFIG" = "xno"; then
-        AC_MSG_ERROR([You need to install pkg-config])
+
+dnl -- The GAC Tool
+
+AC_PATH_PROG(GACUTIL, gacutil, no)
+if test "x$GACUTIL" = "xno" ; then
+	AC_MSG_ERROR([No gacutil tool found])
 fi
 
+AC_SUBST(GACUTIL)
+
+GACUTIL_FLAGS='/gacdir $(DESTDIR)$(libdir)'
+AC_SUBST(GACUTIL_FLAGS)
+
+
 dnl Find Mono and Gmcs
 MONO_REQUIRED_VERSION=1.1.17
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/tapioca-sharp/tapioca/Makefile.am src/tapioca-sharp/tapioca/Makefile.am
--- origsrc/tapioca-sharp/tapioca/Makefile.am	2007-06-08 15:00:59.000000000 -0500
+++ src/tapioca-sharp/tapioca/Makefile.am	2007-06-08 15:11:27.261296200 -0500
@@ -2,8 +2,9 @@
 
 ASSEMBLY_NAME = INdT.Tapioca
 ASSEMBLY = $(ASSEMBLY_NAME).dll
-INdT_Tapiocadir = $(pkglibdir)
+INdT_Tapiocadir = $(libdir)/mono/$(PACKAGE)
 INdT_Tapioca_SCRIPTS = $(ASSEMBLY)
+SNK=$(top_srcdir)/tapioca-sharp.snk
 
 TARGET = $(ASSEMBLY)
 
@@ -45,7 +46,16 @@
 
 
 $(ASSEMBLY): $(SOURCES) $(UTILSOURCES)
-	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $(SOURCES) $(UTILSOURCES) $(PKG_REFERENCES)
+	$(CSC) -target:library -out:$@ $(CSC_FLAGS) $^ $(PKG_REFERENCES) -keyfile:$(SNK)
 
 all: $(ASSEMBLY)
 
+### GAC install rules ###
+
+install-data-local:
+	echo "$(GACUTIL) /i $(ASSEMBLY) $(GACUTIL_FLAGS)"; \
+	$(GACUTIL) /i $(ASSEMBLY) $(GACUTIL_FLAGS) || exit 1;
+
+uninstall-local:
+	echo "$(GACUTIL) /u $(ASSEMBLY_NAME) $(GACUTIL_FLAGS)"; \
+	$(GACUTIL) /u $(ASSEMBLY_NAME) $(GACUTIL_FLAGS) || exit 1;
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/tapioca-sharp/tapioca-sharp.pc.in src/tapioca-sharp/tapioca-sharp.pc.in
--- origsrc/tapioca-sharp/tapioca-sharp.pc.in	2007-06-08 15:01:00.000000000 -0500
+++ src/tapioca-sharp/tapioca-sharp.pc.in	2007-06-08 15:02:48.551531800 -0500
@@ -7,4 +7,4 @@
 Version: @VERSION@
 Cflags:
 Requires: telepathy-sharp
-Libs: -r:${libdir}/tapioca-sharp/INdT.Tapioca.dll
+Libs: -r:${libdir}/mono/tapioca-sharp/INdT.Tapioca.dll
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/tapioca-sharp/tests/Makefile.am src/tapioca-sharp/tests/Makefile.am
--- origsrc/tapioca-sharp/tests/Makefile.am	2007-06-08 15:00:59.000000000 -0500
+++ src/tapioca-sharp/tests/Makefile.am	2007-06-08 15:03:50.989831000 -0500
@@ -22,6 +22,6 @@
 all: $(TARGET)
 
 $(TARGET): $(ASSEMBLY_SOURCES)
-	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $(ASSEMBLY_SOURCES) \
+	$(MCS) $(MCS_FLAGS) $(PKG_REFERENCES) $^ \
 	-target:exe -out:"$@"
 
